src/playground/PlaygroundLoader.tsx

```diff
@@ -301,6 +309,18 @@ function initState(
 		files = defaultPlaygroundState.files;
 	}
 
+	// handle rule domains
+	const ruleDomains: Record<string, string> = {};
+	for (const key of searchParams.keys()) {
+		if (key.startsWith("ruleDomains.")) {
+			const domain = key.slice("ruleDomains.".length);
+			const value = searchParams.get(key);
+			if (value !== null) {
```

Comment:
i believe you did this intentionally, but you might be able to simplify it as follows?:

```suggestion
			if (value) {
```